### PR TITLE
SIP - Sealed Types

### PR DESCRIPTION
**Pull request by dwijnand**
[Rendered](https://github.com/dwijnand/docs.scala-lang/blob/sealed-types/_sips/sips/2020-05-18-sealed-types.md)
[Discussion](https://contributors.scala-lang.org/t/sip-sealed-types/5082)

### CODE REVIEW

File: _sips/sips/2020-05-18-sealed-types.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+opaque type Neg <: Num = Num
+```
+
+The match desugars into an ordinal method, that reuses the logic to associate an ordinal for each
+case:
+
+```scala
+extension (n: Num):
+  def ordinal: Int = (n: Int) match {
+    case n if n > 0 => 0
```
Comment (Jasper-M) [inline _sips/sips/2020-05-18-sealed-types.md:96]: I think the ordinal cases are in the wrong order.

Comment (dwijnand) [reply _sips/sips/2020-05-18-sealed-types.md]: Thank you, fixed.

**Comment by joroKr21** [comment]
Is there a discussion already?

**Comment by lrytz** [comment]
@joroKr21 https://contributors.scala-lang.org/t/sip-sealed-types/5082/12

**Comment by dwijnand** [comment]
Why can't this be merged?

**Comment by SethTisue** [comment]
Sure, we can merge it. But I think the filename should be `2021-05-18-sealed-types.md` (not `2020...`)?

**Pull request merged**